dashboard/src/features/shared/components/SearchBar/SearchBar.jsx

```diff
@@ -83,24 +83,24 @@ class SearchBar extends React.Component {
   handleSubmit(event) {
     event.preventDefault()
 
-    if (this.state.query == this.props.defaultFilter) {
-      this.setState({ showClear: false })
-      this.props.pushList()
-      return
-    }
-
+    const query = {}
     const state = {
       showClear: this.state.query || this.state.sumBy
     }
 
-    const query = {}
-    if (this.state.query) {
-      query.filter = this.state.query
-    } else if (this.props.defaultFilter) {
+    if (this.state.sumBy) query.sum_by = this.state.sumBy
+    if (this.props.defaultFilter) {
       state.query = this.props.defaultFilter
       query.filter = this.props.defaultFilter
     }
-    if (this.state.sum_by) query.sum_by = this.state.sumBy
+
+    if (this.state.query == this.props.defaultFilter) {
+      this.setState({ showClear: false })
```

Comment:
It seems like you can't clear the sum-by field if the filter is equal to the default, but those two should be separate, right? Maybe we can get rid of this entire statement if we change the `state` object to initialize to the following:

```
     const state = {
        showClear: (this.state.query && (this.state.query != this.props.defaultFilter)) || this.state.sumBy
      }
```